src/imp/propagator_dd.js

```diff
@@ -0,0 +1,102 @@
+import _each from '../_each';
+import SpanContextImp from './span_context_imp';
+import CARRIER_BAGGAGE_PREFIX from './propagator_ls';
+
+const CARRIER_DD_TRACER_STATE_PREFIX = 'x-datadog-';
+
+export default class DDProgagator {
+    constructor(tracer) {
+        this._tracer = tracer;
+        this._baggagePrefix = CARRIER_BAGGAGE_PREFIX;
+        this._carrierPrefix = CARRIER_DD_TRACER_STATE_PREFIX;
+    }
+
+    inject(spanContext, carrier) {
+        if (!carrier) {
+            this._tracer._error('Unexpected null carrier in call to inject');
+            return;
+        }
+        if (typeof carrier !== 'object') {
+            this._tracer._error(`Unexpected '${typeof carrier}' FORMAT_TEXT_MAP carrier in call to inject`);
+            return;
+        }
+
+        // eslint-disable-next-line max-len
+        // Per https://github.com/lightstep/lightstep-tracer-javascript/blob/master/src/imp/platform/node/platform_node.js#L91
+        // all generated GUIDs are base 16 strings.
+        // DD headers expect integers, not base 16 values.
+        carrier[`${this._carrierPrefix}parent-id`] = parseInt(spanContext._guid, 16).toString();
```

Comment:
I think it'd be useful to expose a guid getter in the spanContext fwiw (a customer just asked about this)